src/compiler/types/validate-build-package-json.ts

```diff
@@ -4,10 +4,19 @@ import { dirname, join, relative } from 'path';
 import {
   getComponentsDtsTypesFilePath,
   isOutputTargetDistCollection,
+  isOutputTargetDistCustomElements,
   isOutputTargetDistCustomElementsBundle,
   isOutputTargetDistTypes,
 } from '../output-targets/output-utils';
 
+/**
+ * Validate that various fields are set correctly in `package.json`
+ *
+ * @param config the user-supplied Stencil config
+ * @param compilerCtx the compiler context
+ * @param buildCtx the build context
+ * @returns an empty Promise
```

Comment:
oh yeah, I've been adding one just to make it more obvious that it's an async function. But maybe that's already obvious enough?

Reply:
Yeah - I think it comes down to personal preference.  As an example, I personally like really explicit function signatures (even if the return type is `void`/`Promise<void>`) and minimal JSDoc (so omitting `@returns` in the case of those two return types).  But that's super arbitrary of an opinion. At some point we probably should converge on a style guide for this type of thing (as long as there's tooling to support it), but for now I think it's fine as-is.

Reply:
I think maybe I just find 'an empty Promise' to be a bit funny 😄 

I have generally gone for explicit / complete function signatures for a while because I read that supplying them can make typescript run a bit faster (since it doesn't have to infer things, just check that they're correct) but I don't know how true that really is, since it sort of has an urban legend-ey feel to it.

But I'm agreed we should have a standard at some point that we find some way to enforce